PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/utils/PluginUtil.java

```diff
@@ -529,6 +529,29 @@ public static boolean checkPlugInInstallation(String pluginSymbolicName) {
 		return bundles != null && bundles.length >= 0;
 	}
 	
+	/**
+	 * @param targetVersion: Target minimum Java version
+	 * @return Return true if Java version is higher than target. By default return true since only Java 1.7 is required
+	 */
+	public static boolean isJavaVersionHigherThanTarget(double targetVersion) {
+		try {
```

Comment:
Could you try `class JavaVersion` of http://www.eclipse.org/jetty/javadoc/9.4.4.v20170414/org/eclipse/jetty/start/JavaVersion.html, which is already imported and can handle well major/minor version comparison.

Reply:
FYI: http://openjdk.java.net/jeps/223

Reply:
Investigated. Found the package is in org.eclipse.jetty.start. It actually need to download and install. So prefer to keep it this way to avoid introducing the dependency. Will fix after later version of Java.

---
In reply to: [144225783](https://github.com/Microsoft/azure-tools-for-java/pull/1098#discussion_r144225783) [](ancestors = 144225783)

Reply:
I would suggest to refer the code of 
https://github.com/eclipse/jetty.project/blob/3399fd36dc09b0e65bfb840b6e63920f3f1c209b/jetty-util/src/main/java/org/eclipse/jetty/util/JavaVersion.java#L34

Or, refer the class `org.apache.commons.lang3.JavaVersion`